src/keystore.cpp

```diff
@@ -36,6 +36,40 @@ bool CBasicKeyStore::AddKeyPubKey(const CKey& key, const CPubKey &pubkey)
     return true;
 }
 
+bool CBasicKeyStore::HaveKey(const CKeyID &address) const
+{
+    bool result;
```

Comment:
Feel like fixing this gook in a new commit while you're at it?

Just LOCK(cs_KeyStore); return mapKeys.count(address) > 0; should work just as well (there are a few more of these later down).